format-indicator/format-indicator.bal

```diff
@@ -0,0 +1,55 @@
+import ballerina/http;
+
+public type PatientReqV1 record {|
+    "1.0" version = "1.0";
+    string firstName;
+    string lastName;
+    string dob;
+    string diagnosis;
+|};
+
+public type PatientReqV2 record {|
+    "2.0" version = "2.0";
+    Patient patient;
+|};
+
+type PatientReq PatientReqV1|PatientReqV2;  
+
+public type Patient record {|
+    string fullName;
+    string dob;
+    string diagnosis;
+|};
+
+const VERSION_1 = "1.0";
+const VERSION_2 = "2.0";
+
+final http:Client patientClient = check new ("http://api.patients.com.balmock.io");
+
+service /api/v1 on new http:Listener(8080) {
+    
+    resource function post data/patient(PatientReq patintReq) returns error? {
+        VERSION_1|VERSION_2 version = patintReq.version;
+        Patient patient;
+        if version == VERSION_1 {
```

Comment:
Should we do `patintReq is PatientReqV1` instead?
Pros: Ballerina best practice
Const: Hard to see the pattern

Reply:
Let's use `is` since the purpose is to show the Ballerina best practice.